testkit/javadsl/src/main/scala/com/lightbend/lagom/internal/testkit/ProducerStubFactory.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.testkit
+
+import java.util.concurrent.CompletionStage
+import java.util.function.{ Function => JFunction }
+
+import akka.Done
+import akka.actor.{ Actor, ActorRef, Props }
+import akka.stream.javadsl.{ Flow => JFlow, Keep => JKeep, Sink => JSink, Source => JSource }
+import akka.stream.scaladsl.Source
+import akka.stream.{ Materializer, OverflowStrategy }
+import com.lightbend.lagom.javadsl.api.broker.Topic.TopicId
+import com.lightbend.lagom.javadsl.api.broker.{ Subscriber, Topic }
+
+import scala.collection.mutable
+
+private[lagom] class TopicStub[T](topicName: String, topicBuffer: ActorRef, materializer: Materializer) extends Topic[T] {
+  def topicId(): Topic.TopicId = TopicId.of(topicName)
+  def subscribe = new SubscriberStub[T](topicName)(topicBuffer, materializer)
+}
+
+private[lagom] object TopicBufferActor {
+  def props(): Props = Props(new TopicBufferActor())
+  case class SubscribeToBuffer(groupId: String, actorRef: ActorRef)
+}
+
+private[lagom] class TopicBufferActor extends Actor {
+
+  import TopicBufferActor._
+
+  var downstreams = Map.empty[String, ActorRef]
+  val buffer: mutable.Buffer[AnyRef] = mutable.Buffer.empty[AnyRef]
+
+  override def receive: Receive = {
+    case SubscribeToBuffer(groupId, ref) => {
+      downstreams = downstreams + (groupId -> ref)
+      buffer.foreach(msg => ref.tell(msg, ActorRef.noSender))
+    }
+    case message: AnyRef => {
+      downstreams.values.foreach(ref => ref ! message)
+      buffer append message
+    }
+  }
+}
+
+private[lagom] class SubscriberStub[T](topicId: String, groupId: String = "default")(topicBuffer: ActorRef, materializer: Materializer) extends Subscriber[T] {
```

Comment:
This `topicId` a left-over. Will remove it. Thanks for pointing out.
Also, will replace `String` for `TopicId` in related classes. ;-)